src/sql/src/plan/statement/ddl.rs

```diff
@@ -1607,11 +1607,16 @@ fn source_sink_cluster_config(
         (None, None) => Ok(SourceSinkClusterConfig::Undefined),
         (Some(in_cluster), None) => {
             let cluster = scx.catalog.get_cluster(in_cluster.id);
-            if cluster.replica_ids().len() > 1 {
-                sql_bail!("cannot create {ty} in cluster with more than one replica")
-            }
-            if !is_storage_cluster(scx, cluster) {
-                sql_bail!("cannot create {ty} in cluster containing indexes or materialized views");
+
+            if !scx.catalog.system_vars().enable_unified_clusters() {
+                if cluster.replica_ids().len() > 1 {
+                    sql_bail!("cannot create {ty} in cluster with more than one replica")
+                }
```

Comment:
I think you want this to live outside the feature flag check? Things will probably explode in the storage controller if you have multiple replicas of a storage cluster. (You do seem to have a test for this, but I'm surprised that test works.)